dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -772,14 +735,16 @@ void ExchangeReceiverBase<RPCContext>::readLoop(const Request & req)
 
 template <typename RPCContext>
 DecodeDetail ExchangeReceiverBase<RPCContext>::decodeChunks(
-    const RecvMsgPtr & recv_msg,
+    size_t stream_id,
+    const ReceivedMessagePtr & recv_msg,
     std::queue<Block> & block_queue,
     std::unique_ptr<CHBlockChunkDecodeAndSquash> & decoder_ptr)
 {
     assert(recv_msg != nullptr);
     DecodeDetail detail;
 
-    if (recv_msg->chunks.empty())
+    auto & chunks = recv_msg->remaining_consumers == nullptr ? recv_msg->chunks : recv_msg->fine_grained_chunks[stream_id];
```

Comment:
Looks like it can be wrapped into a function.

Reply:
Done